test/slim_check.lean

```diff
@@ -94,3 +99,279 @@ begin
     -- success
   trivial,
 end
+
+open function slim_check
+
+example (f : ℤ → ℤ) (h : injective f) : true :=
+begin
+  have : monotone (f ∘ small.mk),
+  success_if_fail_with_msg
+  { slim_check { random_seed := some 257 } }
+"
+===================
+Found problems!
+
+f := [-1 ↦ 0, -2 ↦ -2, 0 ↦ -1, 1 ↦ 1, x ↦ x]
+x := -1
+y := 0
+guard: -1 ≤ 0 (by construction)
+issue: 0 ≤ -1 does not hold
+-------------------
+",
+  admit,
+  trivial
+end
+
+example (f : ℤ → ℤ) (h : injective f) : true :=
+begin
+  have : monotone f,
+  success_if_fail_with_msg
+  { slim_check { random_seed := some 257 } }
+"
+===================
+Found problems!
+
+f := [-1 ↦ 0, -2 ↦ -2, 0 ↦ -1, 1 ↦ 1, x ↦ x]
+x := -1
+y := 0
+guard: -1 ≤ 0 (by construction)
+issue: 0 ≤ -1 does not hold
+-------------------
+",
+  admit,
+  trivial
+end
+
+section
+
+open tactic
+@[interactive]
+meta def mk_slim_check_test : tactic unit := do
+tgt ← target,
+msg ← (λ s, match interactive.slim_check { random_seed := some 257 } s with
+      | result.success x _ := fail "expecting error" s
+      | result.exception msg _ _ := result.success (msg.iget ()).to_string s
+      end ),
+trace!"Try this:
+  have : {tgt},
+  success_if_fail_with_msg
+  {{ slim_check {{ random_seed := some 257 } }
+\"{msg}\",
+  admit,
+  trivial
+"
+
+end
```

Comment:
Could this move elsewhere so it's not in the middle of tests?

Reply:
I think this is OK. It's analogous to how we're testing `squeeze_simp` after #3947: https://github.com/leanprover-community/mathlib/blob/master/test/squeeze.lean

Reply:
I considered that but it's really only relevant to this file. To get it out of the way, I could write a `test/slim_check_test_cases.lean` file and put it in. How about that?